src/com/spikes2212/genericsubsystems/drivetrains/commands/DriveHolonomicWithPID.java

```diff
@@ -0,0 +1,291 @@
+	 *            command requires and moves.
+	 * @param XSource
+	 *            the {@link PIDSource} this command uses to get feedback for
+	 *            the PID Loop for the X axis.
+	 * @param YSource
+	 *            the {@link PIDSource} this command uses to get feedback for
+	 *            the PID Loop for the Y axis.
+	 * @param setPoint
+	 *            the target point of this command.
+	 *            <p>
+	 *            This command will try to move drivetrain until both axes reach
+	 *            the setpoint. setpoint should be using the same units as
+	 *            drivetrain's {@link PIDSource}s.
+	 *            </p>
+	 * @param PIDSettings
+	 *            the {@link PIDSettings} this command's PIDController needs.
+	 * 
+	 * @see PIDController
+	 */
+	public DriveHolonomicWithPID(HolonomicDrivetrain drivetrain, PIDSource XSource, PIDSource YSource, double setpoint,
+			PIDSettings XPIDSettings, PIDSettings YPIDSettings) {
+		this(drivetrain, XSource, YSource, () -> setpoint, () -> setpoint, XPIDSettings, YPIDSettings);
+	}
+
+	/**
+	 * This constructs a new {@link DriveHolonomicWithPID} using
+	 * {@link PIDSource}s the setpoints for each side, the PID coefficients this
+	 * command's PID loop should have, and the tolerance for error.
+	 *
+	 * @param drivetrain
+	 *            the {@link edu.wpi.first.wpilibj.command.Subsystem} this
+	 *            command requires and moves.
+	 * @param XSource
+	 *            the {@link PIDSource} this command uses to get feedback for
+	 *            the PID Loop for the X axis.
+	 * @param YSource
+	 *            the {@link PIDSource} this command uses to get feedback for
+	 *            the PID Loop for the Y axis.
+	 * @param setPoint
+	 *            a supplier supplying the target point of this command.
+	 *            <p>
+	 *            This command will try to move drivetrain until both axes reach
+	 *            the setpoint. setpoint should be using the same units as
+	 *            drivetrain's {@link PIDSource}s.
+	 *            </p>
+	 * @param PIDSettings
+	 *            the {@link PIDSettings} this command's PIDController needs.
+	 * 
+	 * @see PIDController
+	 */
+	public DriveHolonomicWithPID(HolonomicDrivetrain drivetrain, PIDSource XSource, PIDSource YSource,
+			Supplier<Double> setpoint, PIDSettings XPIDSettings, PIDSettings YPIDSettings) {
+		this(drivetrain, XSource, YSource, setpoint, setpoint, XPIDSettings, YPIDSettings);
+	}
+
+	// Called just before this Command runs the first time
+	protected void initialize() {
+		XMovmentControl = new PIDController(XPIDSettings.getKP(), XPIDSettings.getKI(), XPIDSettings.getKD(), XSource,
+				holonomicDrivetrain::setX);
+		XMovmentControl.setAbsoluteTolerance(XPIDSettings.getTolerance());
+		XMovmentControl.setSetpoint(this.XSetpoint.get());
+		XMovmentControl.setOutputRange(-1, 1);
+		YMovmentControl = new PIDController(YPIDSettings.getKP(), YPIDSettings.getKI(), YPIDSettings.getKD(), YSource,

```

Comment:
Just a small thing, really a matter of style: when doing almost the same thing with different objects (here it's once for x and once for y), it's convenient to put empty new lines (1 or 2) between the separate parts; this usually makes it easier to understand what the function is doing. 

If synchronous operations are important (i.e. calling `enable` for both of them one after the other), then that can also be separated with a blank newline.

Entirely up to you